### PR TITLE
drivers/mrf24j40: add Turbo Mode

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

MRF24J40 supports a proprietary turbo mode with a data rate of 625 kbit/s instead of the standard 250 kbit/s.

It can be enabled through

    ifconfig 7 set high_rate 1

### Testing procedure

Enable High Data rate on a pair of nodes with a mrf24j40 transceiver

```
2020-05-09 16:16:29,865 #  ifconfig 7 set high_rate 1
2020-05-09 16:16:29,869 # success: set high rate on interface 7 to 1
2020-05-09 16:16:32,943 #  ifconfig
2020-05-09 16:16:32,950 # Iface  7  HWaddr: 3B:66  Channel: 26  Page: 0  NID: 0x23  PHY: O-QPSK 
2020-05-09 16:16:32,952 #            high data rate: 1 
2020-05-09 16:16:32,956 #           Long HWaddr: 8F:1B:32:02:AE:20:BB:66 
2020-05-09 16:16:32,963 #            TX-Power: 0dBm  State: IDLE  max. Retrans.: 3  CSMA Retries: 4 
2020-05-09 16:16:32,968 #           ACK_REQ  CSMA  L2-PDU:102 MTU:1280  HL:64  RTR  
2020-05-09 16:16:32,970 #           6LO  IPHC  
2020-05-09 16:16:32,974 #           Source address length: 8
2020-05-09 16:16:32,976 #           Link type: wireless
2020-05-09 16:16:32,982 #           inet6 addr: fe80::8d1b:3202:ae20:bb66  scope: link  VAL
2020-05-09 16:16:32,985 #           inet6 group: ff02::2
2020-05-09 16:16:32,987 #           inet6 group: ff02::1
2020-05-09 16:16:32,991 #           inet6 group: ff02::1:ff20:bb66
2020-05-09 16:16:32,992 #           
2020-05-09 16:16:32,995 #           Statistics for Layer 2
2020-05-09 16:16:32,998 #             RX packets 4  bytes 172
2020-05-09 16:16:33,002 #             TX packets 3 (Multicast: 3)  bytes 129
2020-05-09 16:16:33,006 #             TX succeeded 3 errors 0
2020-05-09 16:16:33,008 #           Statistics for IPv6
2020-05-09 16:16:33,011 #             RX packets 4  bytes 256
2020-05-09 16:16:33,016 #             TX packets 3 (Multicast: 3)  bytes 192
2020-05-09 16:16:33,019 #             TX succeeded 3 errors 0
```

The two nodes can now ping each other, slightly faster than before:

```
2020-05-09 16:18:22,016 #  ping6 fe80::2123:2323:2323:2322 -s 64
2020-05-09 16:18:22,039 # 72 bytes from fe80::2123:2323:2323:2322%7: icmp_seq=0 ttl=64 rssi=-35 dBm time=15.217 ms
2020-05-09 16:18:23,038 # 72 bytes from fe80::2123:2323:2323:2322%7: icmp_seq=1 ttl=64 rssi=-35 dBm time=14.255 ms
2020-05-09 16:18:24,039 # 72 bytes from fe80::2123:2323:2323:2322%7: icmp_seq=2 ttl=64 rssi=-35 dBm time=14.889 ms
2020-05-09 16:18:24,039 # 
2020-05-09 16:18:24,043 # --- fe80::2123:2323:2323:2322 PING statistics ---
2020-05-09 16:18:24,048 # 3 packets transmitted, 3 packets received, 0% packet loss
2020-05-09 16:18:24,052 # round-trip min/avg/max = 14.255/14.787/15.217 ms

2020-05-09 16:18:26,567 #  ifconfig 7 set high_rate 1
2020-05-09 16:18:26,571 # success: set high rate on interface 7 to 1

2020-05-09 16:18:31,008 #  ping6 fe80::2123:2323:2323:2322 -s 64
2020-05-09 16:18:31,024 # 72 bytes from fe80::2123:2323:2323:2322%7: icmp_seq=0 ttl=64 rssi=-36 dBm time=8.313 ms
2020-05-09 16:18:32,026 # 72 bytes from fe80::2123:2323:2323:2322%7: icmp_seq=1 ttl=64 rssi=-36 dBm time=9.913 ms
2020-05-09 16:18:33,027 # 72 bytes from fe80::2123:2323:2323:2322%7: icmp_seq=2 ttl=64 rssi=-35 dBm time=10.873 ms
2020-05-09 16:18:33,027 # 
2020-05-09 16:18:33,031 # --- fe80::2123:2323:2323:2322 PING statistics ---
2020-05-09 16:18:33,036 # 3 packets transmitted, 3 packets received, 0% packet loss
2020-05-09 16:18:33,040 # round-trip min/avg/max = 8.313/9.699/10.873 ms
```

### CODE REVIEW

File: drivers/mrf24j40/mrf24j40_netdev.c [hunk 1]
```diff
@@ -330,6 +330,20 @@ static int _get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len)
             }
             break;
 
+#ifdef MODULE_NETDEV_IEEE802154_OQPSK
+
+        case NETOPT_IEEE802154_PHY:
+            assert(max_len >= sizeof(int8_t));
+            *(uint8_t *)val = IEEE802154_PHY_OQPSK;
+            return sizeof(uint8_t);
```
Comment (leandrolanzieri) [inline drivers/mrf24j40/mrf24j40_netdev.c:338]: Why is this only enabled with `MODULE_NETDEV_IEEE802154_OQPSK`? Is there another option for the phy?

Comment (benpicco) [reply drivers/mrf24j40/mrf24j40_netdev.c:338]: I agree that this is a bit awkward.
I wanted a way to being able to disable it when it's not needed and as the `high_rate` option is only available with `netdev_ieee802154_oqpsk` I thought that's the way to go.

However this is explicitly a non-IEEE802.15.4 mode, so maybe having a separate `netdev_ieee802154_oqpsk_high_rate` pseudo-module would be better here.

File: drivers/mrf24j40/mrf24j40_netdev.c [hunk 2]
```diff
@@ -330,6 +330,20 @@ static int _get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len)
...
+
+        case NETOPT_OQPSK_RATE:
```
Comment (leandrolanzieri) [inline drivers/mrf24j40/mrf24j40_netdev.c:340]: Is this netopt now being used to activate the proprietary mode in radios? From the user perspective wouldn't it be misleading that, if using different radios, when this is set communication does not longer work?

Comment (benpicco) [reply drivers/mrf24j40/mrf24j40_netdev.c:340]: 802.15.4-OQPSK only species 250 kbit/s but many radios implement higher modes. So this netopt is *only* used to activate the proprietary modes.

`NETOPT_MR_OQPSK_RATE` is used to set standard-compliant 802.15.4g MR-OQPSK rates.

File: drivers/mrf24j40/mrf24j40_netdev.c (hunk 1 above)
Comment (leandrolanzieri) [reply drivers/mrf24j40/mrf24j40_netdev.c:338]: In this case `ifconfig` should be changed as well right?

File: drivers/mrf24j40/mrf24j40_netdev.c (hunk 2 above)
Comment (leandrolanzieri) [reply drivers/mrf24j40/mrf24j40_netdev.c:340]: I see. Ok then I'm fine with this, but I think that a follow-up PR which improves the documentation of `NETOPT_OQPSK_RATE` would be good.

File: drivers/mrf24j40/mrf24j40_netdev.c (hunk 1 above)
Comment (benpicco) [reply drivers/mrf24j40/mrf24j40_netdev.c:338]: For conditional compilation, yes.

Comment (leandrolanzieri) [reply drivers/mrf24j40/mrf24j40_netdev.c:338]: I was actually referring to the `NETOPT_IEEE802154_PHY` option. Now that I understand what `NETOPT_OQPSK_RATE` is supposed to do, I think it's OK to enable this functionality just with `MODULE_NETDEV_IEEE802154_OQPSK`.

**Comment by benpicco** [comment]
Thank you for testing!

**Pull request merged**